websites/T/TuneIn/presence.ts

```diff
@@ -1,85 +1,117 @@
 const presence = new Presence({
-		clientId: "844108776793178122",
-	}),
-	strings = presence.getStrings({
+	clientId: "844108776793178122",
+});
+
+async function getStrings() {
+	return presence.getStrings({
 		play: "general.playing",
 		pause: "general.paused",
+		browse: "general.browsing",
 		live: "general.live",
+		listening: "general.listeningMusic",
 	});
+}
 
-let title, author;
+let strings: Awaited<ReturnType<typeof getStrings>>,
+	oldLang: string = null;
 
 presence.on("UpdateData", async () => {
 	const presenceData: PresenceData = {
-		largeImageKey:
-			"https://cdn.rcd.gg/PreMiD/websites/T/TuneIn/assets/logo.png",
-		startTimestamp: Math.floor(Date.now() / 1000),
-	};
-	if (
-		document
-			.querySelector("#innerAppContent")
-			.querySelectorAll('[data-testid="player"]')
-	) {
-		const liveCheck = document.querySelector("#scrubberElapsed");
-		presenceData.largeImageKey =
-			document.querySelector("#playerArtwork").getAttribute("src") ?? "logo";
-		if (!liveCheck) return presence.setActivity();
-		if (liveCheck.textContent === "LIVE") {
-			const pauseCheck = document
-				.querySelector("#innerAppContent")
-				.querySelectorAll('[data-testid="player-status-stopped"]');
-			title = document.querySelector("#playerTitle").textContent;
-			author = document.querySelector("#playerSubtitle").textContent;
+			largeImageKey:
+				"https://cdn.rcd.gg/PreMiD/websites/T/TuneIn/assets/logo.png",
+		},
+		conditions = {
+			isPaused: document.querySelector('[data-testid="player-status-paused"]'),
+			isStopped: document.querySelector(
+				'[data-testid="player-status-stopped"]'
+			),
+			isConnecting: document.querySelector(
+				'[data-testid="player-status-connecting"]'
+			),
+			isLive: document.querySelector("#scrubberElapsed"),
+		},
+		[newLang, timestamps, cover, private] = await Promise.all([
+			presence.getSetting<string>("lang").catch(() => "en"),
+			presence.getSetting<boolean>("timestamps"),
+			presence.getSetting<boolean>("cover"),
+			presence.getSetting<boolean>("privacy"),
+		]);
+
+	if (oldLang !== newLang) {
+		oldLang = newLang;
+		strings = await getStrings();
+	}
+
+	if (!conditions.isConnecting && conditions.isLive) {
+		if (private) presenceData.details = strings.listening;
+		else {
+			const title = document.querySelector("#playerTitle").textContent,
+				author = document.querySelector("#playerSubtitle").textContent,
+				artwork = document.querySelector("#playerArtwork").getAttribute("src");
+
+			if (title) {
+				if (title.length > 128)
+					presenceData.details = `${title.substring(0, 125)}...`;
+				else presenceData.details = title;
+			}
 
-			presenceData.details = title;
-			if (title.length > 128)
-				presenceData.details = `${title.substring(0, 125)}...`;
+			if (author) {
+				if (author.length > 128)
+					presenceData.state = `${author.substring(0, 125)}...`;
+				else presenceData.state = author;
```

Comment:
This kind of stuff shouldn't be needed as its done by the extension 
![image](https://github.com/PreMiD/Presences/assets/34792838/8c79fa81-228b-47e8-a824-78c04c53b368)
If the extension is not doing it correctly please make a post in #feedback.

Reply:
Cool, I didn't know the extension did that, I removed it as requested 👍